RecoEgamma/EgammaTools/src/HGCalClusterTools.cc

```diff
@@ -0,0 +1,56 @@
+#include "RecoEgamma/EgammaTools/interface/HGCalClusterTools.h"
+
+#include "DataFormats/CaloRecHit/interface/CaloCluster.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+float HGCalClusterTools::energyInCone(const float eta,
+                                      const float phi,
+                                      const std::vector<reco::CaloCluster>& layerClusters,
+                                      const float minDR,
+                                      const float maxDR,
+                                      const float minEt,
+                                      const float minEnergy,
+                                      const std::vector<DetId::Detector>& subDets,
+                                      const HGCalClusterTools::EType& eType) {
+  float hadValue = 0.;
+
+  const float minDR2 = minDR * minDR;
+  const float maxDR2 = maxDR * maxDR;
+
+  for (auto& clus : layerClusters) {
+    if (clus.energy() < minEnergy) {
+      continue;
+    }
+
+    if (std::find(subDets.begin(), subDets.end(), clus.seed().det()) == subDets.end()) {
```

Comment:
what's `subDets.size()` on average? If it's more than a few hundred, a more efficient search may be preferrable.

Reply:
The size is small (1-3) as `subDets` stores the 3 the different HGCal subdetector types (HGCEE, HGCSi, HGCSc).